src/main/java/io/phasetwo/service/auth/OrgAddUserAuthenticatorFactory.java

```diff
@@ -57,16 +63,44 @@ private void addUser(AuthenticationFlowContext context) {
       }
       if (!org.hasMembership(context.getUser())) {
         log.infof(
-            "granting membership to %s for user %s",
-            org.getName(), context.getUser().getUsername());
+                "granting membership to %s for user %s",
+                org.getName(), context.getUser().getUsername());
         org.grantMembership(context.getUser());
-        // TODO default roles from config??
+
+        orgs
+                .getUserInvitationsStream(context.getRealm(), context.getUser())
+                .filter(invitationModel -> invitationModel.getOrganization().getId().equals(org.getId()))
+                .findFirst()
```

Comment:
There may technically be more than one invitation for a user to an organization. We should grant them all of the roles from each of the invitations, and revoke all outstanding invitations to that organization.